stage2/modules/fp-basics/README.md

```diff
@@ -1,33 +1,47 @@
+By the end of the module, students will gain insights into applying these concepts to real-world application development, enabling them to develop solutions more adeptly and effectively.
 
-## Приблизительное время прохождения модуля
+## Learning Objectives 🎯
 
-2 часа
+Students will:
 
-## Теория
+- Grasp the concept of pure functions
+- Acquire knowledge of immutability concept, encompassing functions as first-class entities, functions composition, and higher-order functions
+- Delve into recursion
+- Explore currying, partial application, and memoization
 
-- https://doka.guide/tools/fp/
-- https://doka.guide/js/recursion/
-- https://habr.com/ru/company/ruvds/blog/332384/
+## Approximate Module Completion Time ⏱️
 
-## Практика
+- **10 hours**
 
-- Вам необходимо пройти тест "FP-basics" в RS APP > Auto Test
-- Recreate the map function
-- Implement private variable
-- [Codewars](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codewars/Codewars-2022Q1-FP.md)
+## Theory 📖
 
-## Дополнительные материалы
+Students are encouraged to study the following resources:
 
-https://www.youtube.com/watch?v=BMUiFMZr7vk&list=PL0zVEGEvSaeEd9hlmCXrk5yUyqUag-n84
+- [Functional Programming Course](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/learn-about-functional-programming)
+- [What is Functional Programming?](https://www.freecodecamp.org/news/functional-programming-in-javascript/)
+- [What Is a Pure Function in JavaScript?](https://www.freecodecamp.org/news/what-is-a-pure-function-in-javascript-acb887375dfe/)
+- [Mutability vs Immutability](https://www.freecodecamp.org/news/mutability-vs-immutability-in-javascript/)
+- [First-Class and Higher-Order Functions](https://medium.com/@rabailzaheer/first-class-and-higher-order-functions-86d14e40c688#:~:text=Definition%20of%20First%2DClass%20Functions&text=They%20refer%20to%20functions%20that,easily%20accessible%20and%20re%2Dusable.)
+- [The Fundamentals of Function Composition](https://levelup.gitconnected.com/the-fundamentals-of-function-composition-in-javascript-a-complete-guide-with-real-world-examples-9e72a05d1848)
+- [Currying](https://javascript.info/currying-partials)
+- [Currying and Partial Application in JavaScript](https://barker.codes/blog/currying-and-partial-application-in-javascript/)
+- [Recursion and Stack](https://javascript.info/recursion#recursive-structures)
+- [How to use Memoize to cache JavaScript function results and speed up your code](https://www.freecodecamp.org/news/understanding-memoize-in-javascript-51d07d19430e/)
+
+## Practice 💻
+
+- Complete the test "[St2] FP-basics" in the RS-App > Auto Test.
+- Complete ths [Codewars tasks](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codewars/Codewars-2022Q1-FP.md)
```

Comment:
Typo `these/the?`

Reply:
@KristiBo ooops...
Fixed, thank you! ❤️